### PR TITLE
Final recommender experiments

### PR DESCRIPTION
**Pull request by alanefl**
This change prepares the experiment code for evaluating recommenders.  

I'm going to run `sh exp_evaluate_recommenders.sh` on a gcloud instance and let it run for a long time. This will evaluate three recommenders against 4 datasets using 3 settings for the evaluator (so 36 experiments). 

- Fixes some bugs in evaluation/experiment logic that might explain why we were getting 0% for BeerAdvocate.

### CODE REVIEW

File: gbra/recommender/recommenders.py [hunk 1]
```diff
@@ -241,8 +245,10 @@ def _sample_walk_length(self):
         larger alphas bias towards longer walks.
         """
         mu = int(round(self._alpha * self._num_steps_in_walk))
```
Comment (FluxLemur) [inline gbra/recommender/recommenders.py]: maybe rename `num_steps_in_walk` with `max_steps_in_walk`?

File: recommender_experiments/exp_recommender_master.py [hunk 2]
```diff
@@ -3,19 +3,28 @@
...
+# and top_k_recommendations, select N randomly sampled entities
+# to compute the recommender hit ratio on that entity.
+N = 150
 
-recommenders = ["pixie", "popular", "random"]
+recommenders = ["pixie", "random", "popular"]
 
 # The name of the ErdosRenyi graph is of the form
 # ErdosRenyi_[num entities]_[num items]_[num edges]_[graph to draw edge weights from].
 graphs = [
```
Comment (FluxLemur) [inline recommender_experiments/exp_recommender_master.py]: in all these graphs, the first param, min_threshold isn't used at all? did you forget to include it in some computation, or if it's not needed maybe you can remove it?

File: gbra/recommender/recommenders.py (hunk 1 above)
Comment (alanefl) [reply gbra/recommender/recommenders.py]: yeah good point

File: recommender_experiments/exp_recommender_master.py (hunk 2 above)
Comment (alanefl) [reply recommender_experiments/exp_recommender_master.py]: No it is used. It's unpacked in the for-loop below and passed as a command line argument to `exp_recommender.py`

**Comment by alanefl** [comment]
Agree about the parameters of Pixie.  I'll run this, and if results are good, I'm sticking with them. I think the parameters (1000 total with random walks of length 10 in expectation) are pretty reasonable

**Pull request merged**